### PR TITLE
Replace legacy build pipeline with PowerShell SDP build pipeline

### PR DESCRIPTION
**Pull request by jeffreyxrh**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/248668

### CODE REVIEW

**Comment by e0i** [comment]
@traya1 This one seems to come from internal sources. Can you review the changes to the pipeline configuration?

**Comment by e0i** [comment]
@traya1 @DaniHalfin @sdwheeler 

This PR seems to relate to proposed changes to PowerShell Updatable Help pipeline, announced *May* by PowerShell and Docs team. Please review to ensure that the changes are compatible, and do not break anything along the way. Please also do not hesitate to call for help where needed. Thank you.

**Comment by sdwheeler** [comment]
@e0i This is not the change for the updateable help but it is a change to the PowerShell reference rendering pipeline so that PowerShell reference is SDP-based. This is the first step to getting to DocFXv3. You need this update. The build changes the source MD files to YML files. So linking from conceptual docs to this reference must use xref syntax or link to the YML file (even though the source file is still MD).

**Comment by e0i** [comment]
@sdwheeler So can the changes be safely merged, and ignoring the warnings generated above?

**Comment by sdwheeler** [comment]
@e0i The warnings need to be fixed before merging.

**Comment by qinezh** [comment]
Thanks @sdwheeler for the explanation

@e0i We're trying to fix the content issues which cause the build warnings, such like https://github.com/MicrosoftDocs/windows-powershell-docs/pull/2088, and there may be still some remaining content issues like invalid file links, we'll let you know then you could fix them before we switch to the new SDP build pipeline.

**Comment by e0i** [comment]
Closing and re-opening to refresh checks as #2094, #2095, and #2096 have been merged.

File: docset/windows/Microsoft.Windows.ServerManager.Migration/Export-SmigServerSetting.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ description: Use this topic to help manage Windows and Windows Server technologi
 external help file: Microsoft.Windows.ServerManager.PowerShell.dll-Help.xml
 keywords: powershell, cmdlet
 manager: jasgro
-ms.date: 2017-01-03
+ms.date: 01/03/2017
```
Comment (qinezh) [inline docset/windows/Microsoft.Windows.ServerManager.Migration/Export-SmigServerSetting.md:10]: can we rollback the changes if it's not needed?

**Comment by qinezh** [comment]
@e0i can you please take a look at this PR? It's ready to merged, here's the preview page: https://review.docs.microsoft.com/en-us/powershell/module/webadministration/add-webconfigurationlock?view=win10-ps&branch=sdp

**Comment by qinezh** [comment]
@e0i @traya1 @DaniHalfin This PR is the prerequisite to switch to DocFX v3, please help to take a look.

**Comment by qinezh** [comment]
Any updates? @e0i @traya1 @DaniHalfin 

cc @rloutlaw

**Comment by e0i** [comment]
@qinezh Thank you for following up. We are working to fix the build warnings before merge as suggested by Sean.

**Comment by qinezh** [comment]
@e0i Can you please share with us the progress of fixing these content issues?

Currently almost all the PowerShell reference repos are migrated to use the new SDP build pipeline, and are ready for the docfx v3 rollout. At the same time, the legacy build pipeline will be retired very soon. 

cc @rloutlaw

**Comment by e0i** [comment]
@qinezh  Thank you for following up with this. Incorrect links are being removed to fix the build warnings and the PR is going to be merged soon.

**Comment by rloutlaw** [comment]
@e0i - Can you review and merge this PR? The permissions have changed in this repo from the default, so I want to avoid adjusting my own permissions (as an organization admin) just to merge this PR.

**Comment by traya1** [comment]
closing/reopening for new validation checks

**Comment by rloutlaw** [comment]
@traya1 @e0i  Can you provide and update to this PR? I'm a PM on the Docs Design/Engineering team and we need to get this change in as part of our feature rollout.

**Comment by traya1** [comment]
Not sure if this helps but the warnings in the PR call out yml files while the topics in the repo are md. Is this a possible build issue?

**Comment by qinezh** [comment]
Hi @traya1 , this is a known issue and it will be resolved after we enabled docfx v3. And this PR is the prerequisite step of docfx v3 rollout.

**Comment by e0i** [comment]
@qinezh Thank you for the clarification. Independent contributors have been working to fix the build warnings and we have been able to take out warnings that stem from zero guid articles so far.

Given your statement, can we explicitly confirm that it is OK to continue merging this PR with the remaining build warnings, as those relate to a known issue that is going to be fixed in v3?

**Comment by qinezh** [comment]
@ei0, the remaining build warnings are due to invalid file links, which is the content issue. So we have to fix the content.

The issue mentioned by @traya1 is that, the source files in build report is **Yaml file** instead of the correct **markdown file**. That will be resolved in docfx v3.

**Comment by e0i** [comment]
@qinezh Thank you for the clarification. We then continue to fix the build warnings immediately. Expect sign off by Monday UTC.

**Comment by qinezh** [comment]
@ei0, I'd recommend to merge this PR if the only concern from your side is the build warnings. Since you could get a more accurate build repo generated by the new SDP build pipeline after this PR merged, then you could make sure that all the invalid file links are fixed in following PRs. Thoughts?

**Comment by e0i** [comment]
Marking the PR with the SignOff label based on the suggestion above by @qinezh, which I think makes sense because we have already processed the 400+ build warnings and have created a PR that fixes 200+, and rest of the fixes are under way.

cc @traya1

**Comment by e0i** [comment]
Closing and reopening to refresh checks.

**Comment by rjagiewich** [comment]
We are merging this with approval from @Dansimp on the assumption that all the build warnings will be fixed, per the comment above.

**Pull request merged**